client/devdocs/design/search-collection.jsx

```diff
@@ -31,49 +34,106 @@ const shouldShowInstance = ( example, filter, component ) => {
 	return ! filter || searchPattern.toLowerCase().indexOf( filter ) > -1;
 };
 
-const Collection = ( { children, filter, section = 'design', component } ) => {
-	let showCounter = 0;
-	const summary = [];
-
-	const examples = React.Children.map( children, example => {
-		if ( ! example || ! shouldShowInstance( example, filter, component ) ) {
-			return null;
-		}
-
-		const exampleName = getComponentName( example );
-		const exampleLink = `/devdocs/${ section }/${ camelCaseToSlug( exampleName ) }`;
-
-		showCounter++;
-
-		if ( filter ) {
-			summary.push(
-				<span key={ `instance-link-${ showCounter }` } className="design__instance-link">
-					<a href={ exampleLink }>{ exampleName }</a>
-					,&nbsp;
-				</span>
+class Collection extends React.PureComponent {
+	static propTypes = {
+		children: PropTypes.node,
+		component: PropTypes.string,
+		examplesToMount: PropTypes.number,
+		filter: PropTypes.string,
+		section: PropTypes.string,
+	};
+
+	static defaultProps = {
+		examplesToMount: 7,
+		section: 'design',
+	};
+
+	_setLazyRenderRef = ref => {
+		this._lazyRender = ref;
+	};
+
+	componentDidMount = () => {
+		this._lazyRenderTimeout = setTimeout( () => {
+			// HACK: Reach into the <LazyRender> component and force all
+			// examples to be mounted after a few seconds of the component
+			// being mounted. This means if the user loads the page and
+			// changes context the rest of the content won't wait to load
+			// until they scroll down.
```

Comment:
I am not sure I understand the trade-off here – is the hack (good job at dissecting how the lazy render component works) worth the wait once we change the context? Also, what are some UX scenarios when somebody changes the context, I am failing to imagine that.

Reply:
I guess I should clarify the comment, it's not even changing contexts (switching tabs, changing apps, etc.), but simply *not* scrolling down past the first "X" number of elements within the first few seconds. If the user loads the page and reads a few components, then scrolls down, this could fire. It could be that it's too generous and a timeout of something like `400ms` is more appropriate though.

It basically means if the page isn't scrolled past the loaded content right away we load the rest in the background, transparent to the user, so they aren't left waiting later when we can just load it now.